src/applications/caregivers/components/PreSubmitInfo/index.jsx

```diff
@@ -3,56 +3,38 @@ import React, { useEffect, useState } from 'react';
 import SignatureCheckbox from './SignatureCheckbox';
 
 const PreSubmitCheckboxGroup = ({ onSectionComplete, formData, showError }) => {
-  const veteranLabel = `Enter Veteran\u2019s full name`;
-  const primaryLabel = `Enter Primary Family Caregiver applicant\u2019s full name`;
-  const secondaryOneLabel = `Enter Secondary Family Caregiver applicant\u2019s full name`;
-  const secondaryTwoLabel = `Enter Secondary Family Caregiver applicant\u2019s (2) full name`;
-
-  const [signatures, setSignature] = useState({
-    [veteranLabel]: false,
-    [primaryLabel]: false,
+  const veteranLabel = `Veteran\u2019s`;
+  const primaryLabel = `Primary Family Caregiver applicant\u2019s`;
+  const secondaryOneLabel = `Secondary Family Caregiver applicant\u2019s`;
+  const secondaryTwoLabel = `Secondary Family Caregiver (2) applicant\u2019s`;
+  const createInputContent = label => `Enter ${label} full name`;
+  const hasSecondaryOne = formData['view:hasSecondaryCaregiverOne'];
+  const hasSecondaryTwo = formData['view:hasSecondaryCaregiverTwo'];
+
+  const [signatures, setSignatures] = useState({
+    [createInputContent(veteranLabel)]: false,
+    [createInputContent(primaryLabel)]: false,
   });
 
-  const [secondaryCaregivers, setSecondaryCaregivers] = useState({
-    [secondaryOneLabel]: false,
-    [secondaryTwoLabel]: false,
-  });
   const unSignedLength = Object.values(signatures).filter(
     obj => Boolean(obj) === false,
   ).length;
 
+  // when there is no unsigned signatures set AGREED (onSectionComplete) to true
+  // if goes to another page (unmount), set AGREED (onSectionComplete) to false
   useEffect(
     () => {
-      if (!unSignedLength) {
-        onSectionComplete(true);
-      }
-
-      if (unSignedLength) {
-        onSectionComplete(false);
-      }
+      if (!unSignedLength) onSectionComplete(true);
 
-      const hasSecondaryOne =
-        formData?.secondaryOneFullName?.first &&
-        formData?.secondaryOneFullName?.last;
+      if (unSignedLength) onSectionComplete(false);
```

Comment:
I think this would look cleaner with the above lines combined into `onSectionComplete(!unSignedLength);`

Reply:
Good looks!